cognite/client/data_classes/data_modeling/views.py

```diff
@@ -392,19 +390,54 @@ def as_apply(self) -> MappedPropertyApply:
 
 
 @dataclass
-class ConnectionDefinition(ViewProperty):
-    ...
+class ConnectionDefinition(ViewProperty, ABC):
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        if "connectionType" not in resource:
+            raise ValueError(f"{cls.__name__} must have a connectionType")
+        connection_type = to_snake_case(resource["connectionType"])
+
+        if connection_type in {"multi_edge_connection", "single_edge_connection"}:
+            return cast(Self, SingleHopConnectionDefinition.load(resource))
+        elif connection_type in {"single_reverse_direct_relation", "multi_reverse_direct_relation"}:
+            return cast(Self, ReverseSingleHopConnection.load(resource))
+        else:
+            raise ValueError(f"Cannot load {cls.__name__}: Unknown connection type {connection_type}")
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        raise NotImplementedError
 
 
 @dataclass
 class SingleHopConnectionDefinition(ConnectionDefinition):
+    """Describes the edge(s) that are likely to exist to aid in discovery and documentation of the view.
+    A listed edge is not required. i.e. It does not have to exist when included in this list.
+    A connection has a max distance of one hop.
+
+    It is called 'EdgeConnection' in the API spec.
+
+    Args:
+        type (DirectRelationReference): Reference to the node pointed to by the direct relation. The reference
+            consists of a space and an external-id.
+        source (ViewId): The target node(s) of this connection can be read through the view specified in 'source'.
+        name (str | None): Readable property name.
+        description (str | None): Description of the content and suggested use for this property.
+        edge_source (ViewId | None): The edge(s) of this connection can be read through the view specified in
+            'edgeSource'.
+        direction (Literal["outwards", "inwards"]): The direction of the edge. The outward direction is used to
+            indicate that the edge points from the source to the target. The inward direction is used to indicate
+            that the edge points from the target to the source.
+        connection_type (Literal["single_edge_connection", "multi_edge_connection"]): The type of connection,
+            either a single or multi edge connections are expected to exist.
+    """
+
     type: DirectRelationReference
     source: ViewId
     name: str | None = None
     description: str | None = None
     edge_source: ViewId | None = None
     direction: Literal["outwards", "inwards"] = "outwards"
-    connection_type: Literal["multiEdgeConnection"] = "multiEdgeConnection"
+    connection_type: Literal["single_edge_connection", "multi_edge_connection"] = "multi_edge_connection"
```

Comment:
Let's remove `connection_type` from these classes and just have one class per type connection type instead.

Reply:
done